lib/fs/filesystem.go

```diff
@@ -22,23 +22,38 @@ type Filesystem interface {
 	DirNames(name string) ([]string, error)
 	Lstat(name string) (FileInfo, error)
 	Mkdir(name string, perm FileMode) error
+	MkdirAll(name string, perm FileMode) error
```

Comment:
Could we reduce the API surface by having just one of mkdir/mkdirall (preferably the latter)?